notion/renderer.py

```diff
@@ -0,0 +1,241 @@
+import markdown2
+import requests
+
+from .block import *
+
+
+class BaseRenderer(object):
+
+	def __init__(self, start_block):
+		self.start_block = start_block
+
+	def render(self):
+		return self.render_block(self.start_block)
+
+	def calculate_child_indent(self, block):
+		if block.type == "page":
+			return 0
+		else:
+			return 1
+
+	def render_block(self, block, level=0, preblock=None, postblock=None):
+		assert isinstance(block, Block)
+		type_renderer = getattr(self, "handle_" + block._type, None)
+		if not callable(type_renderer):
+			if hasattr(self, "handle_default"):
+				type_renderer = self.handle_default
+			else:
+				raise Exception("No handler for block type '{}'.".format(block._type))
+		pretext = type_renderer(block, level=level, preblock=preblock, postblock=postblock)
+		if isinstance(pretext, tuple):
+			pretext, posttext = pretext
+		else:
+			posttext = ""
+		return pretext + self.render_children(block, level=level+self.calculate_child_indent(block)) + posttext
```

Comment:
Is level always going to be either 0 or 1? Can you ever have a `page` in a `page` type block?